src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -179,25 +179,7 @@ private File installOrUpdateScanner(String dirPath) throws IOException, Integrat
         
         final HttpUrl downloadUrl = buildDownloadUrl();
         final String currentInstalledVersion = determineInstalledVersion().orElse("");
-        final Response response = download(installDirectory, downloadUrl, currentInstalledVersion);
-        String newVersionString = response.getHeaderValue(DOWNLOAD_VERSION_HEADER);
-        String newFileName;
-        if (newVersionString == null && (newFileName = response.getHeaderValue(DOWNLOADED_FILE_NAME)) != null) {
-            newVersionString = getVersionFromDetectFileName(newFileName);
-        }
-        if (response.isStatusCodeSuccess() && newVersionString != null) {
-            logger.info("{} Old version: {}, New Version from header: {}", LOG_PREFIX, currentInstalledVersion, newVersionString);
-            if (StringUtils.isNotBlank(newVersionString)
-                && !newVersionString.equals(currentInstalledVersion)
-                && !isDownloadVersionTooOld(currentInstalledVersion, newVersionString)) {
-                return handleSuccessResponse(response, installDirectory.getAbsolutePath(), newVersionString);
-            }
-        } else if (response.getStatusCode() == 304) {
-            logger.info("{} Present Detect installation is up to date - skipping download.", LOG_PREFIX);
-        } else {
-            logger.warn("{} Unable to download artifact. Response code: {} {}", LOG_PREFIX, response.getStatusCode(), response.getStatusMessage());
-        }
-        return null;
```

Comment:
This method is refactored so that response is not returned from try-with block before file had been downloaded.